core/src/main/java/com/epam/eco/kafkamanager/core/KafkaManagerImpl.java

```diff
@@ -567,31 +570,48 @@ public void deletePermissions(ResourcePermissionsDeleteParams params) {
-                                    params.getFilter().getResourceType(),
-                                    params.getFilter().getResourceName(),
-                                    params.getFilter().getPatternType()));
-                        }
+                params.getFilter(), permissionsToDelete -> {
+                    if (params.getFilter().getPrincipalObjectFilter() == null) {
+                        return;
+                    }
+
+                    List<PermissionInfo> allResourcePermissions =
+                            permissionRepo.findMatchingOfResource(params.getFilter().toMatchAnyFilter());
+                    if (allResourcePermissions.size() == permissionsToDelete.size()) {
+                        metadataRepo.delete(
+                                PermissionMetadataKey.with(
+                                params.getFilter().getPrincipalObjectFilter(),
+                                params.getFilter().getResourceType(),
+                                params.getFilter().getResourceName(),
+                                params.getFilter().getPatternType()));
                     }
                 });
     }
 
     @Override
     public void deletePermissions(PrincipalPermissionsDeleteParams params) {
-        throw new UnsupportedOperationException();
+        List<PermissionInfo> permissions = getPrincipalPermissions(params.getPrincipalObject());
+        Set<ResourcePermissionFilter> filters = permissions.stream()
+                                               .filter(permissionInfo ->
+                                                       !params.containsInExcludes(permissionInfo.getResourceName(),
+                                                                                  permissionInfo.getResourceType())
+                                                       && params.matchResourceType(permissionInfo.getResourceType()))
+                                               .map(permissionInfo -> ResourcePermissionFilter.builder()
+                                                               .resourceType(permissionInfo.getResourceType())
+                                                               .resourceName(permissionInfo.getResourceName())
+                                                               .patternType(permissionInfo.getPatternType())
+                                                               .principalFilter(params.getPrincipalObject())
+                                                               .build())
+                                               .collect(Collectors.toSet());
+
+        filters.forEach(filter -> deletePermissions(new ResourcePermissionsDeleteParams(filter)));
+
+    }
+
+    private List<PermissionInfo> getPrincipalPermissions(KafkaPrincipal kafkaPrincipal) {
+        return getAllPermissions().stream()
+                                  .filter(permission -> permission.getKafkaPrincipal().equals(kafkaPrincipal))
```

Comment:
filtering could have happened on **permissionRepo** side and create appropriate method like getAllPermissions(kafkaPrincipal)?

Reply:
Yes. Problem here is: you cant delete record from permission (acl list) without knowing it recourceName. And the reason of getting all permissions - to know the resource name and type.